### PR TITLE
Fix Java version output

### PR DESCRIPTION
**Pull request by adam-thorpe**
Closes: https://github.com/AdoptOpenJDK/openjdk-build/issues/1511
FYI @jerboaa 

Signed-off-by: Adam Thorpe <adam.thorpe@ibm.com>

### CODE REVIEW

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -57,12 +57,12 @@ class Build {
     Integer getJavaVersionNumber() {
         // version should be something like "jdk8u" or "jdk" for HEAD
         def matcher = (buildConfig.JAVA_TO_BUILD =~ /(\d+)/)
-        if (matcher.matches()) {
+        if (matcher.find()) {
             List<String> list = matcher[0] as List
             return Integer.parseInt(list[1] as String)
         } else if ("jdk".equalsIgnoreCase(buildConfig.JAVA_TO_BUILD.trim())) {
             // This needs to get updated when JDK HEAD version updates
```
Comment (jerboaa) [inline pipelines/build/common/openjdk_build_pipeline.groovy]: With your update to use `buildConfig.JAVA_FEATURE_VERSION` we should remove that comment as it's no longer a hard-coded value.

**Comment by jerboaa** [comment]
> Nice work! Thanks for fixing this and sorry for the breakage.
> 
> There is a second hard-coded value here:
> https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/openjdk_build_pipeline.groovy#L65
> 
> Could you use `buildConfig.JAVA_FEATURE_VERSION` there too, please? Thank you!

Err, I meant:
`pipelines/build/common/build_base_file.groovy:211`

It's fine to not do that, in this PR though. Removing the comment in `openjdk_build_pipeline.groovy` would be sufficient.

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (jerboaa) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: On second thought. Will `JAVA_FEATURE_VERSION` be available at this point. Looks to me that there are only these parameters going to be generated:
https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/create_job_from_template.groovy#L64

**Comment by jerboaa** [comment]
Perhaps @johnoliver should look this over.

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (adam-thorpe) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: I'm not too sure, I didn't realise only this list of values we're passed into the groovy scripts. Surely just adding JAVA_FEATURE_VERSION to this list would be a good idea as we don't want too many hardcoded values that we'd need to change in 6 months time

Comment (adam-thorpe) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Or adding it as a constant somehow?

**Comment by adam-thorpe** [comment]
@jerboaa I've reverted the change so that it is just the regex fix as I would like this to get fixed asap so we can get some testing done in the nightlies tonight. I'll create an issue for the magic numbers

**Pull request merged**